mono/mini/driver.c

```diff
@@ -1941,6 +1941,14 @@ mono_main (int argc, char* argv[])
 		} else if (strcmp (argv [i], "--nacl-null-checks-off") == 0){
 			nacl_null_checks_off = TRUE;
 #endif
+		} else if (strncmp (argv [i], "--assembly-loader=", strlen("--assembly-loader=")) == 0) {
```

Comment:
One nice thing about this approach is we could add a third "warn" option at some point that emits stderr spew every time it detects we're loading an assembly of the wrong strongname.